core/build.gradle.kts

```diff
@@ -59,44 +45,23 @@ kotlin {
 publishing {
     publications {
         create<MavenPublication>("coreJVM") {
-            groupId = "codes.laurence.warden"
-            artifactId = "warden-core"
-            version = version
+            artifact(tasks.jvmJar)
```

Comment:
I set the artifacts to be based on task inputs, so Gradle will automatically be clever and trigger those tasks if and only if required.